R/compute_quantiles.R

```diff
@@ -0,0 +1,62 @@
+
+#' @title Calculate Quantiles for Rates and Cumulative Hazard
+#' @description Calculate quantiles for samples of rates and cumulative hazard 
+#' outputted from \link[threemc]{circ_sample_tmb}, and add them as columns to 
+#' the shell `data.frame` `out` with estimated empirical circumcision rates.
+#' 
+#' @param out Shell dataset with a row for every unique record in 
+#' circumcision survey data for a given area. Also includes empirical estimates
+#'  for circumcision estimates for each unique record.
+#' @param fit Object containing `samples` list for the (cumulative) incidence 
+#' and hazard rate of circumcision for the region(s) in question. 
+#' @param probs Specific quantiles to be calculated,
+#' Default: c(0.5, 0.025, 0.975)
+#' @param names Parameter with \link[stats]{quantile}: logical; if true, the 
+#' result has a names attribute. Set to FALSE for speedup with many probs,
+#' Default: FALSE
+#' @param ... Further arguments passed to or from other methods.
+#' @return Input `out` `data.frame`, including columns with quantiles for 
+#' hazard rates etc for different circumcision types, and for overall 
+#' circumcision.
+
+#' @seealso 
+#'  \code{\link[threemc]{circ_sample_tmb}}
+#'  \code{\link[stats]{quantile}}
+#' @rdname compute_quantiles
+#' @export
+
+# function to calculate quantiles for rates and cumulative hazard
+# maybe add "textProgressBar" to for loop if it's too slow?
+compute_quantiles <- function(out, 
+                              fit, 
+                              probs = c(0.5, 0.025, 0.975), 
+                              names = FALSE, ...) {
+  
+  probs <- sort(probs)
+  
+  # function to do "legwork" of computing quantiles
```

Comment:
Is Jeff going to work on this package? If so he will probably ask you to use double ##, I think single # means something special to emacs which causes it to format these lines weirdly

Reply:
Yeah, Emacs Speaks Statistics default indenting centers lines that are commented with a single `#`. I'm sure there is a way to change that, but I'm not clever enough to figure out how.

Reply:
I think you can do something like `(setq ess-indent-with-fancy-comments nil)` in your emacs config Jeff, but I've changed all the comments on their own lines to have (at least) two #s